src/FloatingActionButton/FloatingActionButton.spec.js

```diff
@@ -0,0 +1,31 @@
+/* eslint-env mocha */
+import React from 'react';
+import {shallow} from 'enzyme';
+import {assert} from 'chai';
+
+import FloatingActionButton from './FloatingActionButton';
+import getMuiTheme from '../styles/getMuiTheme';
+import ContentAdd from '../svg-icons/content/add';
+
+describe('<FloatingActionButton />', () => {
+  const muiTheme = getMuiTheme();
+  const shallowWithContext = (node) => shallow(node, {context: {muiTheme}});
+
+  describe('hover state', () => {
+    it('should reset the hover state when disabled', () => {
+      const wrapper = shallowWithContext(
+        <FloatingActionButton>
+          <ContentAdd />
+        </FloatingActionButton>
+      );
+      wrapper.setState({
```

Comment:
If I run this test:

```js
it('should reset the hover state when disabled', () => {
  const wrapper = shallowWithContext(
    <FloatingActionButton>
      <ContentAdd />
    </FloatingActionButton>
  );
  // wrapper.setState({
  //   hovered: true,
  // });
  wrapper.children().at(0).simulate('mouseEnter');
});
```

This exception has been thrown:

```
<FloatingActionButton /> hover state should reset the hover state when disabled:
     TypeError: Cannot read property 'isKeyboardFocused' of undefined
      at FloatingActionButton._this.handleMouseEnter (FloatingActionButton.js:204:10)
      at node_modules/enzyme/build/ShallowWrapper.js:768:23
      at ReactDefaultBatchingStrategyTransaction.Mixin.perform (node_modules/react/lib/Transaction.js:138:20)
      at Object.ReactDefaultBatchingStrategy.batchedUpdates (node_modules/react/lib/ReactDefaultBatchingStrategy.js:63:19)
      at batchedUpdates (node_modules/react/lib/ReactUpdates.js:98:20)
      at node_modules/enzyme/build/ShallowWrapper.js:767:45
      at withSetStateAllowed (node_modules/enzyme/build/Utils.js:196:3)
      at ShallowWrapper.simulate (node_modules/enzyme/build/ShallowWrapper.js:764:42)
      at Context.<anonymous> (FloatingActionButton.spec.js:24:32)
```

This is because `handleMouseEnter` method sets state conditionally by checking `isKeyboardFocused` on `EnhancedButton`. Reference to `EnhancedButton` is retrieved using `ref`. If you run `npm run lint` you will get warning for this line:

```
204:10  warning  Using this.refs is deprecated                          react/no-string-refs
```